### ISSUE TITLE
loads of errors: "Request textDocument/foldingRange failed with message: Cannot read properties of undefined (reading 'lineFoldingOnly')"

### ISSUE THREAD

**Issue by szechp** [AstroNvim/AstroNvim#1680]
### Checklist

- [X] Using a stable version of Neovim (i.e. not neovim nightly)
- [X] `:AstroUpdate`
- [X] Restarted AstroNvim

### Operating system/version

ubuntu 22.04

### Terminal/GUI

gnome

### AstroNvim Health

==============================================================================
astronvim: require("astronvim.health").check()

AstroNvim ~
- AstroNvim Version: v3.4.2
- Neovim Version: v0.9.0-dev-1086+gaa840ab56
- WARNING Neovim nightly is not officially supported and may have breaking changes
- OK `git` is installed: Used for core functionality such as updater and plugin management
- OK `xdg-open` is installed: Used for `gx` mapping for opening files with system opener (Optional)
- OK `lazygit` is installed: Used for mappings to pull up git TUI (Optional)
- OK `node` is installed: Used for mappings to pull up node REPL (Optional)
- WARNING `gdu` is not installed: Used for mappings to pull up disk usage analyzer (Optional)
- WARNING `btm` is not installed: Used for mappings to pull up system monitor (Optional)
- OK `python` is installed: Used for mappings to pull up python REPL (Optional)

(i downgraded to stable, same error)

### Describe the bug

when i'm in an ansible playbook i get loads of errors like this: 
```
Error executing vim.schedule lua callback: UnhandledPromiseRejection with the reason:
RPC[Error] code_name = InternalError, message = "Request textDocument/foldingRange failed with message: Cannot read properties of undefined (reading 'lineFoldingOnly')"
```

### Steps to Reproduce

1. open yaml
2. error everywhere

### Expected behavior

no errors

**Comment by mehalter** [comment]
What language servers are you using for Ansible playbooks?

**Comment by szechp** [comment]
2 client(s) attached to this buffer: 
 
 Client: null-ls (id: 1, bufnr: [100, 25, 3])
 	filetypes:       css, markdown.mdx, javascriptreact, html, graphql, less, scss, typescript, javascript, markdown, json, jsonc, handlebars, yaml, vue, typescriptreact, sh, python, luau, lua
 	autostart:       false
 	root directory:  /home/philipp/repos/kubernetes-of-cloud
 	cmd:             <function>

**Comment by szechp** [comment]
Nope I installed it via the commands: lspinstall, reinstall etc.

**Comment by mehalter** [comment]
Ok, `LspInstall` installs through Mason.  What version do you have installed? You can check with `:Mason` and using the UI to see more information about `ansible-language-server`

**Comment by mehalter** [comment]
Is it possible for you to share the ansible file/project that you are working with? Just to see if I can replicate it? Also once the errors pop up, does it look like `ansiblels` has crashed? Can you run `:LspLog`? Does this happen with every ansible playbook?

It seems like a bug in the language server since that's what's erroring out. This is almost certainly an upstream bug.

**Comment by szechp** [comment]
> Ok, `LspInstall` installs through Mason. What version do you have installed? You can check with `:Mason` and using the UI to see more information about `ansible-language-server`

    ✓ ansible-language-server
      Ansible Language Server
      
      installed version 1.0.4                                              
      homepage          https://github.com/ansible/ansible-language-server 
      languages         Ansible                                            
      categories        LSP                                                
      executables       ansible-language-server 

ansiblels doesnt crash, it works just fine. it starts when i save tough, im beginning to think maybe it has something to do with the linter? i disabled linting on save, but it still pop up...

**Comment by mehalter** [comment]
Can you share an example project/file that you can replicate this with? Can you replicate it without your user configuration and just opening a clean installation and installing the Ansible language server with mason?

**Comment by szechp** [comment]
okay it even happens with yamlls:
i opened a hosts.yaml file it looks like this
"name":
  "hosts":
    "some-name":
      "ansible_host": "1.2.3.4"
      "ansible_user": "root"
    "some-other-name":
      "ansible_host": "5.6.7.8"
      "ansible_user": "root"

**Comment by mehalter** [comment]
```sh
mv ~/.config/nvim/lua/user ~/.config/nvim/lua/user.bak
mv ~/.local/share/nvim ~/.local/share/nvim.bak
mv ~/.local/state/nvim ~/.local/state/nvim.bak
mv ~/.cache/nvim ~/.cache/nvim.bak
```

if you are using neovim nightly v0.9 then you can do this:

```sh
git clone https://github.com/AstroNvim/AstroNvim ~/.config/stock_astronvim
NVIM_APPNAME=stock_astronvim nvim
```

I cannot replicate it with that `hosts.yaml` with `yamlls` either

**Comment by szechp** [comment]
okay i found out it comes from an astrocommunity package.

**Comment by szechp** [comment]
so after some digging, i found there is a issue with yaml treesitter and folding. when i disable folding, no errors are shown.

**Comment by mehalter** [comment]
Can you elaborate? Can you replicate with a stock installation without any user configuration?

**Comment by szechp** [comment]
https://github.com/kevinhwang91/nvim-ufo/issues/72 theres the issue

**Comment by mehalter** [comment]
We are doing this. Can you replicate without any user configuration and just default AstroNvim?

**Comment by szechp** [comment]
> We are doing this. Can you replicate without any user configuration and just default AstroNvim?

yes i can. adding the above code to polish fixes the error.

**Comment by mehalter** [comment]
Adding the above code to your polish function would not change anything regarding how the LSP is setup. It is just setting a variable to a value that's then never used.


https://github.com/AstroNvim/AstroNvim/blob/main/lua/astronvim/utils/lsp.lua#L336

Here is where we set it for all language servers

**Comment by szechp** [comment]
> 

haha okay yes i figured it out now, lsp just wasnt laoded, thats why i didnt get the error. its still there... but it has something to do with ufo

**Comment by mehalter** [comment]
I cannot replicate this with a default installation of AstroNvim without any user configuration with just the yaml language servers installed.

**Comment by mehalter** [comment]
Can you share replication instructions from a completely clean environment to recreating the error? Make sure this clean environment has refreshed data/state/config folders to make sure there aren't lingering old files

**Comment by mehalter** [comment]
Yeah, sadly I cannot replicate this with a fresh installation and following these instructions. I have verified all of the `nvim-ufo` settings are correct and cannot get nvim-ufo to complain. Maybe want to make sure you have cleared your cache/state/data folders there might be something lingering that is making it freak out.

**Comment by szechp** [comment]
so after disabling all and enabling everything one by one, i finally found the culprit: its this plugin: https://github.com/someone-stole-my-name/yaml-companion.nvim

(i have this in my yaml.lua, which why I always installed it on a fresh install, without realising it. sorry for the confusion)

**Comment by mehalter** [comment]
No worries, definitely when testing on a fresh, default installation make sure to not have any user configuration or anything running. Also for what it's worth. We are doing schema validation automatically in AstroNvim with schemastore if you are using yamlls. So you probably don't need this anyway.
